ignite/handlers/state_param_scheduler.py

```diff
@@ -0,0 +1,342 @@
+            plt.xlabel("events")
+            plt.ylabel("values")
+            plt.legend()
+
+        """
+        keys_to_remove = ["save_history"]
+        for key in keys_to_remove:
+            if key in scheduler_kwargs:
+                del scheduler_kwargs[key]
+        values = []
+        scheduler = cls(save_history=False, **scheduler_kwargs)
+        engine = Engine(lambda e, b: None)
+        for i in range(num_events):
+            scheduler(engine=engine)
+            values.append([i, getattr(engine.state, scheduler_kwargs["param_name"])])
+        return values
+
+
+class LambdaStateScheduler(StateParamScheduler):
+    """Update a parameter during training by using a user defined function.
+        User defined function is taking an event index as input and returns parameter value.
+
+    Args:
+        lambda_fn: user defined parameter update function.
+        param_name: name of parameter to update.
+        save_history: whether to log the parameter values to
+            `engine.state.param_history`, (default=False).
+
+    Examples:
+
+        .. code-block:: python
+
+            ...
+            engine = Engine(train_step)
+
+            initial_value = 10
+            gamma = 0.99
+
+            lambda_scheduler = LambdaStateScheduler(
+                param_name="lambda",
+                lambda_fn=lambda event_index: initial_value * gamma ** (event_index % 9),
+            )
+
+            lambda_state_parameter_scheduler.attach(engine, Events.EPOCH_COMPLETED)
+
+            # basic handler tp print scheduled state parameter
+            engine.add_event_handler(Events.EPOCH_COMPLETED, lambda _ : print(engine.state.custom_scheduled_param))
+
+            engine.run([0] * 8, max_epochs=2)
+
+    .. versionadded:: 0.6.0
+
+    """
+
+    def __init__(self, lambda_fn: Callable, param_name: str, save_history: bool = False):
+        super(LambdaStateScheduler, self).__init__(param_name, save_history)
+        self.lambda_fn = lambda_fn
+        self._state_attrs += ["lambda_fn"]
+
+    def get_param(self) -> Union[List[float], float]:
+        return self.lambda_fn(self.event_index)
+
+
+class LinearStateScheduler(StateParamScheduler):
+    """Update a parameter during training by using linear function.
+    The function keeps the parameter value to zero until step_zero steps passed and then linearly increases it to 1
+    until an additional step_one steps passed. Continues the trend until it reaches max_value.
+
+    Args:
+        initial_value : starting value of the parameter.
+        step_constant : step index until parameter's value is kept constant.
+        step_max_value : step index at which parameter's value becomes max_value.
+        max_value : max parameter value.
+        param_name: name of parameter to update.
+        save_history: whether to log the parameter values to
+            `engine.state.param_history`, (default=False).
```

Comment:
The name of the class and the docs are unclear about what this scheduler does. There are no `step_zero`, `step_one` arguments anymore.

Reply:
`PiecewiseLinearStateScheduler` would be maybe more appropriate ?  But we should implement the same behavior as it is done for `PiecewiseLinear` ( from `ParamScheduler`)